requirements/optional.txt

```diff
@@ -0,0 +1,16 @@
+# some of these dependencies live on a private git server. If you have the address,
+# configure git with:
+# git config --global url.https://<private hostname>/.insteadOf https://private.repo/
```

Comment:
@rosskarchner Ok. Thinking long-term, doing it the way you're proposing here might encourage more things to be public and open source than private anyway.

I'm sold on this for now. Let's have one source of truth, and let this be it.

:+1: